### PR TITLE
*: Fix ddl test and inadvertently modify column info.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by zxylvlp** [comment]
Which op will change? Cast will?

File: session_test.go [hunk 1]
```diff
@@ -1689,6 +1690,18 @@ func (s *testSessionSuite) TestGlobalVarAccessor(c *C) {
 	c.Assert(v, Equals, varValue2)
 }
 
+func (s *testSessionSuite) TestColumnInfoModified(c *C) {
+	store := newStore(c, s.dbName)
```
Comment (qiuyesuifeng) [inline session_test.go]: Should we close the store at the end of the test?

Comment (coocood) [inline session_test.go]: It doesn't matter.

**Comment by coocood** [comment]
@zxylvlp Yes it is. 
It doesn't create a new field type when changing it.
https://github.com/pingcap/tidb/blob/coocood/fix-ddl-and-column-modified/optimizer/typeinferer.go#L70

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by coocood** [comment]
@zxylvlp PTAL

**Comment by zxylvlp** [comment]
LGTM

**Pull request merged**